### DISCUSSION TITLE
Nuxt Layers/Extends Support

### DISCUSSION THREAD

**Discussion by warflash** [wobsoriano/trpc-nuxt#101] [Q&A]
Hey 👋

I'm wondering if there's a way to use this module together with the nuxt layers feature: https://nuxt.com/docs/guide/going-further/layers

Things like having an auth/user router in the base repository and then plugging individual routers into the base router on the project that extends the base repo.

Maybe this is already possible but unfortunately I couldn't figure out how, so if someone got resources or examples for that I'd highly appreciate it!

**Comment by arily** [comment]
If the scopes is too big I think letting us create multiple endpoints could be a temporary solution.

But then we need a way to choose which endpoint we are hitting, so instead of app.$client we may need a composable like useTRPC(route: string): ?<- different routers depends on given route.

**Reply by wobsoriano** [reply]
I haven't really tried with layers yet, but based on your suggestion about creating multiple endpoints, you could create different plugins pointing to different endpoints instead of `useTRPC(route: string)`?
